zerver/models.py

```diff
@@ -1481,6 +1481,14 @@ def __str__(self) -> str:
         return f"<Recipient: {display_recipient} ({self.type_id}, {self.type})>"
 
 
+# Does save the foreign key of a external authentication
+# source to allow the users email to change
+class ExtAuthId(models.Model):
+    id: int = models.AutoField(auto_created=True, primary_key=True, verbose_name="ID")
+    ldap: Optional[str] = models.TextField(null=True)
+    saml: Optional[str] = models.TextField(null=True)
```

Comment:
It's not obvious to me that this has a `user_id` field? Is that what `id` here represents? We should also probably be using `BigAutoField` with modern conventions.

Reply:
If I remember correctly, the Idea is that the ExtAuthId object contains the IDs of several external auth providers the user has "linked".

So, ldap contains the unique ID on the LDAP side.
ID should be the link to the Zulip User object

Reply:
Yeah, I get the idea, but I think it might be nicer to have this table just be a standard table, maybe named `LinkedExternalAccounts`, with rows of the form `id/user_id/auth_backend_name/external_id`.

The `auth_backend_name` IDs would be the slugs we unique -- the `name` field from `dict_representation` that's the unique string ID for that backend used in the API (and thus hopefully stable). That would make it correct for organizations where, say, there are multiple SAML providers -- we'd have the IdP name in the `auth_backend_name`.

I think that'd be a lot more extensible than this model where there's only one row per user.